sky/server/rest.py

```diff
@@ -90,26 +91,51 @@ def decorator(func: F) -> F:
         @functools.wraps(func)
         def wrapper(*args, **kwargs):
             backoff = common_utils.Backoff(initial_backoff, max_backoff_factor)
-            with _retry_in_context():
-                for retry_cnt in range(max_retries):
+            failed_retry_cnt = 0
```

Comment:
Maybe call it `consecutive_failed_count` or something? I would not expect this one to to get reset to 0 if we make progress